sentry/src/main/java/io/sentry/instrumentation/file/SentryFileInputStream.java

```diff
@@ -127,28 +128,42 @@ public static final class Factory {
     public static FileInputStream create(
         final @NotNull FileInputStream delegate, final @Nullable String name)
         throws FileNotFoundException {
-      return new SentryFileInputStream(
-          init(name != null ? new File(name) : null, delegate, ScopesAdapter.getInstance()));
+      final @NotNull IScopes scopes = ScopesAdapter.getInstance();
+      return isTracingEnabled(scopes)
+          ? new SentryFileInputStream(init(name != null ? new File(name) : null, delegate, scopes))
+          : delegate;
     }
 
     public static FileInputStream create(
         final @NotNull FileInputStream delegate, final @Nullable File file)
         throws FileNotFoundException {
-      return new SentryFileInputStream(init(file, delegate, ScopesAdapter.getInstance()));
+      final @NotNull IScopes scopes = ScopesAdapter.getInstance();
+      return isTracingEnabled(scopes)
+          ? new SentryFileInputStream(init(file, delegate, scopes))
+          : delegate;
     }
 
     public static FileInputStream create(
         final @NotNull FileInputStream delegate, final @NotNull FileDescriptor descriptor) {
-      return new SentryFileInputStream(
-          init(descriptor, delegate, ScopesAdapter.getInstance()), descriptor);
+      final @NotNull IScopes scopes = ScopesAdapter.getInstance();
+      return isTracingEnabled(scopes)
+          ? new SentryFileInputStream(init(descriptor, delegate, scopes), descriptor)
+          : delegate;
     }
 
     static FileInputStream create(
         final @NotNull FileInputStream delegate,
         final @Nullable File file,
         final @NotNull IScopes scopes)
         throws FileNotFoundException {
-      return new SentryFileInputStream(init(file, delegate, scopes));
+      return isTracingEnabled(scopes)
+          ? new SentryFileInputStream(init(file, delegate, scopes))
+          : delegate;
+    }
+
+    private static boolean isTracingEnabled(final @NotNull IScopes scopes) {
```

Comment:
you can just use `ScopesAdapter.getInstance()` here, without passing it as parameter, as it is always the same.
Same for `SentryFileOutputStream`

Reply:
nope, we gotta use the parameter one because we pass a mocked one in tests.

Reply:
i don't get it
You pass `ScopesAdapter.getInstance()` when you call this function, and there's no need to pass a Scopes in the tests.
Or are you going to add a new test?

Reply:
we pass it here right? https://github.com/getsentry/sentry-java/pull/4039/files#diff-aca0bdac62da1139f1a2196522ff6b0d33f9e700797b88b6c8de04f6bd14b134L72

Reply:
i missed that one 😅